ordering/__init__.py

```diff
@@ -61,6 +61,18 @@ def compare(self, left_item: T, right_item: T) -> bool:
 
         return self._labels[left_item] < self._labels[right_item]
 
+    def replace(self, existing_item: T, new_item: T) -> None:
```

Comment:
Yes to the assert. I was just focusing on the core logic.

A pity about the repeated asserts. I'm not too worried about it, but if you have an idea, I'd like to hear it.